Hardware/Hardware/CardReader/PaymentIntent.swift

```diff
@@ -30,5 +30,96 @@ public extension PaymentIntent {
     /// Metadata Keys
     enum MetadataKeys {
         public static let store = "paymentintent.storename"
+
+        /// The customer's name - first name then last name separated by a space, or
+        /// empty if neither first name nor last name is given.
+        /// This key is also used by the plugin when it creates payment intents.
+        ///
+        public static let customerName = "customer_name"
+
+        /// The customer's email address, or empty if not given.
+        /// This key is also used by the plugin when it creates payment intents.
+        ///
+        public static let customerEmail = "customer_email"
+
+        /// The store URL, e.g. `https://mydomain.com`
+        /// This key is also used by the plugin when it creates payment intents.
+        ///
+        public static let siteURL = "site_url"
+
+        /// The order ID, e.g. 6140
+        /// This key is also used by the plugin when it creates payment intents.
+        ///
+        public static let orderID = "order_id"
+
+        /// The order key, e.g. `wc_order_0000000000000`
+        /// This key is also used by the plugin when it creates payment intents.
+        ///
+        public static let orderKey = "order_key"
+
+        /// The payment type, i.e. `single` or `recurring`
+        /// See also PaymentIntent.PaymentTypes
+        /// This key is also used by the plugin when it creates payment intents.
+        ///
+        public static let paymentType = "payment_type"
+    }
+}
+
+public extension PaymentIntent {
+    enum PaymentTypes {
+        /// A payment that IS NOT associated with an order containing a subscription
+        /// This key is also used by the plugin when it creates payment intents.
+        ///
+        public static let single = "single"
+
+        /// A payment that IS associated with an order containing a subscription
+        /// This key is also used by the plugin when it creates payment intents.
+        ///
+        public static let recurring = "recurring"
+    }
+}
+
+// MARK: - Convenience Initializers
+//
+public extension PaymentIntent {
+    static func initMetadata(store: String? = nil,
```

Comment:
Just for kicks and giggles, this could be a good candidate for a [builder](https://en.wikipedia.org/wiki/Builder_pattern). Not sure it's worth the effort though

Reply:
I was hoping there was a Swift-ier way. I'll look at builder for a future PR? Thankfully the unit tests will catch if/when a builder doesn't build it the same way :)

Reply:
I think this looks fine as it is, but if you are looking for something more Swifty, you could have a Metadata struct that then converts to a dictionary. Also, to simplify, it's OK to assign a `nil` value to a dictionary: it will clear the value for that key if it exists, or it will be ignored if it doesn't.

```swift
public extension PaymentIntent {
    struct Metadata {
        let store: String?
        let customerName: String?
        let customerEmail: String?
        //...
        
        var asDictionary: [String: Any] {
            var metadata = [String: Any]()
            metadata[Keys.store] = store
            metadata[Keys.customerName] = customerName
            metadata[Keys.customerEmail] = customerEmail
            // ...
            return metadata
        }

        /// Metadata Keys
        enum Keys {
            public static let store = "paymentintent.storename"
            public static let customerName = "customer_name"
            public static let customerEmail = "customer_email"
            // ...
        }
    }
}
```

A possibly swiftier way would be to use `Codable` and a [dictonary encoder](https://elegantchaos.com/2018/02/21/decoding-dictionaries-in-swift.html), but bringing in a new dependency or writing our own encoder seems too much for something like this.

Reply:
> it's OK to assign a nil value to a dictionary: it will clear the value for that key if it exists, or it will be ignored if it doesn't

Removed unneeded `nil` checks in https://github.com/woocommerce/woocommerce-ios/pull/4208/commits/b76acdc243e9a1a251f8ad0f2a22b3aab6f2022f